### ISSUE TITLE
Failure to recognize CpuCredits flag in launch template

### ISSUE THREAD

**Issue by Yashiroo** [terraform-providers/terraform-provider-aws#5761]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
Terraform v0.11.7
provider.aws: version = "~> 1.33"

### Affected Resource(s)
aws_autoscaling_group
aws_launch_template

### Terraform Configuration Files

```
resource "aws_launch_template" "lt_reverse_proxy" {
  name_prefix            = "${format("lt-%s-reverse-proxy-", var.env)}"
  image_id               = "${data.aws_ami.amznami.id}"
  instance_type          = "${var.instanceType}"
  key_name               = "${format("kp-%s-%s", var.env, var.servicesKeypair)}"
  vpc_security_group_ids = ["${data.aws_security_group.reverse_proxy_sg.id}", "${data.aws_security_group.tools_sg.id}"]
  user_data              = "${base64encode(data.template_file.user_data.rendered)}"

  iam_instance_profile = {
    name = "${data.aws_iam_instance_profile.reverse_proxy_instance_profile.name}"
  }

  credit_specification {
    cpu_credits = "${var.cpuCreditsMode}"
  }

  lifecycle {
    create_before_destroy = true
  }

}

```

### Debug Output
```
Action=CreateLaunchTemplate&ClientToken=xxxx&LaunchTemplateData.IamInstanceProfile.Name=xxxx&LaunchTemplateData.ImageId=xxxx&LaunchTemplateData.InstanceType=t3.micro&LaunchTemplateData.KeyName=kp-a-services&LaunchTemplateData.SecurityGroupId.1=sg-xxxx&LaunchTemplateData.SecurityGroupId.2=sg-xxxx&LaunchTemplateData.UserData=xxxx&LaunchTemplateName=lt-x-reverse-proxy-xxxx&Version=2016-11-15
```

### Expected Behavior
EC2 instance launched from this Launch Template should have the cpucredits mode set to `standard`.

### Actual Behavior
EC2 instances have the cpucredits mode set to "unlimited".

### Steps to Reproduce

1. Create launch template and set `cpu_credits` to "standard"
2. Create an Autoscaling Group that uses this Launch Template
3. Start instances using the Autoscaling Group

### Important Factoids
I have tried to check the log in debug but it was too verbose, however i have not seen `CreditSpecification` or `CpuCredits` anywhere in the logs.
I'm guessing terraform recognizes the changes as shown below but it does not implement them?
```
CREATE: aws_launch_template.lt_reverse_proxy
arn:                                "" => "<computed>"
credit_specification.#:             "" => "1"
credit_specification.0.cpu_credits: "" => "standard"
default_version:                    "" => "<computed>"
iam_instance_profile.#:             "" => "1"
iam_instance_profile.0.name:        "" => "xxx"
image_id:                           "" => "ami-xxx"
instance_type:                      "" => "t3.micro"
key_name:                           "" => "kp-a-services"
latest_version:                     "" => "<computed>"
name:                               "" => "<computed>" (forces new resource)
name_prefix:                        "" => "lt-x-reverse-proxy-" (forces new resource)
user_data:                          "" =>
```

**Comment by bflad** [comment]
I have submitted a pull request to enable T3 instance types to configure `credit_specification` within the `aws_launch_template` resource: #5922

**Comment by bflad** [comment]
Support has been merged into master and will release with version 1.38.0 of the AWS provider, likely middle of next week. 👍

**Comment by bflad** [comment]
This has been released in [version 1.38.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1380-september-26-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
